### ISSUE TITLE
Varia: Logo misalignment on grid-based headers

### ISSUE THREAD

**Issue by eduardozulian** [Automattic/themes#1467]
## Steps to replicate

1. Activate a theme with a grid-based header layout. Example: Alves or Morden
2. Go to Customize > Site Identity
3. Add a logo
4. Hide the site title/tagline
5. Visit the site

### With site title, without logo
<img width="1109" alt="Captura de Tela 2019-10-02 às 12 19 01" src="https://user-images.githubusercontent.com/1136160/66058288-7f549d80-e510-11e9-8b56-cf9ca6c24b74.png">

### With site title and logo
<img width="1072" alt="Captura de Tela 2019-10-02 às 12 19 11" src="https://user-images.githubusercontent.com/1136160/66058331-94313100-e510-11e9-8c18-f58d9dfda9e5.png">

### Without site title, with logo (and fewer menu items for a better understanding)

<img width="1077" alt="Captura de Tela 2019-10-02 às 12 19 57" src="https://user-images.githubusercontent.com/1136160/66058365-a4491080-e510-11e9-832d-2b1d254f6e53.png">

## Expected

To have my logo horizontally aligned with my menu. When I choose to not display my site title, I'm assuming that it's redundant and that we place it where the site title should be.

A quick workaround would be to change the grid template when we have a logo:

```
@media only screen and (min-width: 560px) {
	.wp-custom-logo .site-header {
		grid-template-areas:
        	"site-logo main-navigation"
        	"site-title ."
        	"site-description social-navigation";
  }	
}
```

<img width="1048" alt="Captura de Tela 2019-10-02 às 12 21 15" src="https://user-images.githubusercontent.com/1136160/66059644-ba57d080-e512-11e9-85e0-1079ed6d3b1f.png">

***

User report: none, #dogfooding

**Comment by eduardozulian** [comment]
The same happens with Morden so I believe that all themes using a grid in the header are probably experiencing the same problem.

Report on p9cu9o-1Xr-p2

**Comment by andreabadgley19** [comment]
Giving this a +1. A customer using the Alves theme in 16714749-hc wanted the logo horizontally aligned with the menu. I gave them the workaround you suggested, @eduardozulian, and created a followup at 2498815-z if we need to reach out about it again.

**Comment by samiff** [comment]
Used the CSS workaround in 16848776-hc

**Comment by eduardozulian** [comment]
Another report: 17655404-hc

**Comment by ccwalburn** [comment]
Another user wanted their logo aligned with the menu using the Hever theme: 17656893-hc
Used the CSS workaround from @eduardozulian

**Comment by GeoJunkie** [comment]
Also encountered during a Quick Start session with Morden theme. CSS sent in 2579322-zen.

**Comment by caimin-j** [comment]
I had this is chat too. Sent the CSS code in 2724903-zen.

**Comment by tbradsha** [comment]
18802202-hc. CSS hotfix above worked.

**Comment by tavilla61** [comment]
Another user who wants their logo aligned with the main nav menu. 
CSS given in chat: #19279868-hc.

**Comment by poojashetty18** [comment]
Another report: 19783020-hc with Morden theme

**Comment by mikeicode** [comment]
Another request 2823309-zen

**Comment by kjellr** [comment]
@allancole @danieldudzic @jeffikus: is this a bug, or is this how the them was originally intended to behave?

**Comment by mikeicode** [comment]
Also requested 20043356-hc - Alves theme

**Comment by jeffikus** [comment]
@kjellr yeah, I've been wondering about this myself as I tried it on Hever a few days ago. As far as I can tell this was the original behavior, but @allancole would be the best person to confirm. I've been using custom CSS to move the grid positioning around -> https://gist.github.com/jeffikus/27a3218ccd58bd045788c8d363eb5a2a and adjusting the margin depending on how big the logo is.

**Comment by ktyfuller604** [comment]
Requested here - [20133022-hc](https://mc.a8c.com/support-stats/happychat/transcript.php?id=20133653) 
Sent them the CSS

**Comment by mikeicode** [comment]
Requested 20543275-hc

**Comment by kjellr** [comment]
It looks like we can fix this by conditionally changing the `grid-template-areas` of `.site-header` from... 

```
grid-template-areas: "site-logo site-logo" "site-title main-navigation" "site-description social-navigation";
```

... to something like ... 

```
grid-template-areas: "site-logo main-navigation" "site-logo social-navigation";
```

... depending on whether the title is shown or not. We could do this easily if we added a body class or something to hook into. @allancole / @jeffikus do you happen to know how to get the value of that setting? Some google searching is not coming up with anything that works. 🤔

**Comment by Lochlaer** [comment]
Also reported by one of our DIFM clients. Used CSS to fix.

**Comment by mikeicode** [comment]
CSS provided 24178124-hc

**Comment by bsanevans** [comment]
Adding 6 more reports here, originally reported on pbEMJv-bj-p2

2880539-zd-woothemes (Morden)
3305374-zd-woothemes (Hever)
3289631-zd-woothemes (Morden)

3310598-zd-woothemes (Hever – plan had no CSS option to apply fix)
3301663-zd-woothemes (Alves – plan had no CSS option to apply fix)

**Comment by fashuvo** [comment]
Another report in 25214291-hc.

User had the Morden theme and I provided the CSS workaround.

**Comment by scruffian** [comment]
I'm reluctant to do this at a theme level. Some users will have a logo at the top and the site title/tagline turned off. Making this change will change their sites without them doing anything which seems risky. Thoughts?

**Comment by kjellr** [comment]
Yeah, I know what you mean. I wish this had been implemented more cleanly initially. It would be worth getting some HE feedback on whether they think this would negatively effect folks.

**Comment by allancole** [comment]
The misalignment here was a simple oversight—not intentional at all. 

Looking at the number of HE’s who’ve already reported on this issue, it might be worth it to risk disrupting sites toward something that clearly looks better and intentional :-). 

I like @kjellr’s suggestion [here](https://github.com/Automattic/themes/issues/1467#issuecomment-620000060). It’s pretty easy to do a conditional class that checks for `get_theme_mod( 'custom_logo' );` and adds a class to the `<body>`.

**Comment by scruffian** [comment]
I have a fix for this here: https://github.com/Automattic/themes/pull/2782

**Comment by inaikem** [comment]
We have an affected customer back in contact to say the issue has not been resolved for them: 25689867-HC and 3483324-ZD. Thanks!

**Comment by eduardozulian** [comment]
@MaggieCabrera, I'm reopening as this is still an issue on 4897250-zen . Can we confirm if the fix is also being applied to Atomic sites?

For now, I share the same CSS workaround.

**Comment by jordesign** [comment]
Closing as there's been no further follow up (or other reports) in almost 3 years
